### PR TITLE
mDNS server bootstrap

### PR DESCRIPTION
**Pull request by celebdor**
Runs mDNS server on the cluster, first on bootstrap and then on the masters

### CODE REVIEW

**Comment by celebdor** [comment]
(contains #89 patches until they get merged, then I'll rebase).

**Comment by celebdor** [comment]
also TODO:
- missing remove the SRV entries from libvirt baremetal network

**Comment by derekhiggins** [comment]
Can one of the admins verify this patch?

**Comment by danehans** [comment]
As we discussed over Slack, running mDNS as a [static pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) can provide significant benefits over the approach taken by this PR. By following the static pod approach, mDNS will essentially be managed like other k8s control-plane components. For example:

- [Bootkube](https://github.com/kubernetes-incubator/bootkube) can be used to launch mDNS.
- A unique Ignition config for mDNS will no longer be required.
- Keepalived/VRRP will no longer be required.
- Multiple mDNS pods can be load-balanced by a k8s service IP.
- Systemd is not longer required for mDNS.
- Setup scripts can be reduced or eliminated through [ConfigMaps](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) and [container envs](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/).

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/100/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/102/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/105/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/109/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/114/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/120/

File: 06_deploy_bootstrap_vm.sh [hunk 1]
```diff
@@ -43,6 +43,79 @@ $GOPATH/src/github.com/openshift/installer/bin/openshift-install --dir ocp --log
...
+    interface \${INTERFACE}
+    virtual_router_id 51
+    priority 50
+    advert_int 1
+    authentication {
+        auth_type PASS
+        auth_pass cluster_uuid_api_vip
+    }
+    virtual_ipaddress {
+        \${MASTER_VIP}
```
Comment (cybertron) [inline 06_deploy_bootstrap_vm.sh]: This also needs "scope link" added to it.

Comment (cybertron) [reply 06_deploy_bootstrap_vm.sh]: Or maybe not. I was actually missing the scope on the API_VIP for the master nodes below too. Not sure whether it matters on the bootstrap.

**Comment by celebdor** [comment]
Closed in favor of #171

**Pull request closed without merging**